include/dsn/cpp/json_helper.h

```diff
@@ -214,11 +214,11 @@
     }
 
 #define NON_MEMBER_JSON_SERIALIZATION(type, ...)                                                   \
-    inline void json_encode(dsn::json::JsonWriter &output, const type &t)                          \
+    void json_encode(dsn::json::JsonWriter &output, const type &t)                                 \
```

Comment:
Why remove inline?

Reply:
In cpp, inline function must be defined in `.h`, not `.cpp`. And we should move `json_code` function of `gpid` to `gpid.cpp`.